enable multi-cards relate test case (#886)
DeepLink-org/deeplink.framework: dipu/torch_dipu/csrc_dipu/CMakeLists.txt

set(GENERATED_KERNELS_VENDOR "${PROJECT_SOURCE_DIR}/third_party/DIOPI/impl/${UsedVendor}/convert_config.yaml")
set(GENERATED_KERNELS_SCRIPT "${AUTOGEN_DIOPI_WRAPPER_DIR}/autogen_diopi_wrapper.py")
set(GENERATED_KERNELS_CONFIG "${AUTOGEN_DIOPI_WRAPPER_DIR}/diopi_functions.yaml")

> #set(DEVICE_GUARD_FREE_VENDOR cuda) # vensors that do not need to device guard need to be added to this list.

Reviewer: ```suggestion
#set(DEVICE_GUARD_FREE_VENDORS cuda) # Vendors that do not require Device Guard should be included in this list..
```

Reviewer: cuda 是不是，不是的话把它直接变成空，比注释掉好理解
如果要新加一个 vendor，会不知道要不要保留 cuda

Reviewer 2: 补充个信息, 之前 diopi torch 里是调用的 aten api, 里面会有自己的 guard.
但是这次 优化后, 很多算子直接调 native api了. native 没有执行guard. 后面类cuda的 应该都没法 忽略 guard了.

Author: 实际上原来也需要，因为在调用aten之前，wrapper 里面也有memory 、 stream、 event相关操作
